src/Microsoft.DotNet.SignTool/src/VerifySignatures.cs

```
@@ -16,6 +19,53 @@ namespace Microsoft.DotNet.SignTool
 {
     internal class VerifySignatures
     {
+        internal static bool VerifySignedDeb(string filePath)
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                // We cannot check the signature of a .deb file on Windows.
+                return true;
+            }
+
+            string tempDir = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
+            Directory.CreateDirectory(tempDir);
+
+            // https://microsoft.sharepoint.com/teams/prss/esrp/info/SitePages/Linux%20GPG%20Signing.aspx
+            try
+            {
+                // Download the Microsoft public key
+                using (HttpClient client = new HttpClient())
+                {
+                    using (Stream stream = client.GetStreamAsync("https://packages.microsoft.com/keys/microsoft.asc").Result)
+                    {
+                        using (FileStream fileStream = File.Create($"{tempDir}/microsoft.asc"))
+                        {
+                            stream.CopyTo(fileStream);
+                        }
+                    }
+                }
+
+                RunCommand($"ar x {filePath} --output {tempDir}");
+                RunCommand($"gpg --import {tempDir}/microsoft.asc");
+                RunCommand($"cat {tempDir}/debian-binary {tempDir}/control.tar.gz {tempDir}/data.tar.gz > {tempDir}/combined-contents");
+                string output = RunCommand($"gpg --verify {tempDir}/_gpgorigin {tempDir}/combined-contents");
+                if (output.Contains("Good signature"))
+                {
+                    return true;
+                }
+                
+                return false;
+            }
+            catch(Exception)
+            {
+                return false;
+            }
```

This looks questionable. Do we really need to swallow exceptions completely without doing any logging?

---

Good point. I'll log them. I think it's still fair to return false in this instance though because the signtool will fail appropriately when verification fails for a file that is expected to be signed.

---

Addressed in https://github.com/dotnet/arcade/pull/15216/commits/05ecf2c13a54d7d0ddb0aef2ba4a62cbbba19462